pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentGenerationJobRunner.java

```diff
@@ -281,7 +317,11 @@ private void submitSegmentGenTask(File localTempDir, URI inputFileURI, int seqId
           _outputDirFS.copyFromLocalFile(localSegmentTarFile, outputSegmentTarURI);
         }
       } catch (Exception e) {
-        LOGGER.error("Failed to generate Pinot segment for file - {}", inputFileURI, e);
+        String msg = "Failed to generate Pinot segment for file - " + inputFileURI.toString();
+        _failure.compareAndSet(null, new RuntimeException(msg, e));
+
+        // We have to decrement the latch by the number of pending tasks.
```

Comment:
Suggest shutting down the executor service in the main thread after all the tasks are submitted. It can throw exception if we shut it down here but still trying to submit another task.
We can drain the latch here by checking the remaining count:
```
        long count = _segmentCreationTaskCountDownLatch.getCount();
        for (int i = 0; i < count; i++) {
          _segmentCreationTaskCountDownLatch.countDown();
        }
```